actors/builtin/miner/miner_test.go

```diff
@@ -1414,10 +1414,59 @@ func TestDeclareRecoveries(t *testing.T) {
 		// Attempt to recover
 		dlIdx, pIdx, err := st.FindSector(rt.AdtStore(), oneSector[0].SectorNumber)
 		require.NoError(t, err)
-		rt.ExpectAbortContainsMessage(exitcode.ErrInsufficientFunds, "does not cover pledge requirements", func () {
-			actor.declareRecoveries(rt, dlIdx, pIdx, bf(uint64(oneSector[0].SectorNumber)))
+		rt.ExpectAbortContainsMessage(exitcode.ErrInsufficientFunds, "does not cover pledge requirements", func() {
+			actor.declareRecoveries(rt, dlIdx, pIdx, bf(uint64(oneSector[0].SectorNumber)), big.Zero())
 		})
 	})
+
+	t.Run("recovery must pay back fee debt", func(t *testing.T) {
```

Comment:
This is great. It would be nice if the new functionality in `PreCommitSector`, `ProveCommitSector`, `ConfirmSectorProofsValid` and `WithdrawBalance` was also covered.

Reply:
Yes, I will add a test for `PreCommitSector` and `WithdrawBalance`.  `ProveCommitSector` verification is going away (#961) so I'll drop that one.  I don't think this PR changes `ConfirmSectorProofsValid` but let me know if I'm missing a hidden change trickling from precommit.